src/pip/_internal/build_env.py

```diff
@@ -57,14 +79,10 @@ def _create_standalone_pip() -> Generator[str, None, None]:
         return
 
     with TempDirectory(kind="standalone-pip") as tmp_dir:
-        pip_zip = os.path.join(tmp_dir.path, "__env_pip__.zip")
-        kwargs = {}
-        if sys.version_info >= (3, 8):
-            kwargs["strict_timestamps"] = False
-        with zipfile.ZipFile(pip_zip, "w", **kwargs) as zf:
-            for child in source.rglob("*"):
-                zf.write(child, child.relative_to(source.parent).as_posix())
-        yield os.path.join(pip_zip, "pip")
+        pip_runner = os.path.join(tmp_dir.path, "__pip-runner__.py")
+        with open(pip_runner, "w") as f:
```

Comment:
```suggestion
        with open(pip_runner, "w", encoding="utf8") as f:
```

Reply:
Good catch! The temp directory could easily contain Unicoide characters, and even in 3.11, `open` still doesn't default to UTF-8.